File: src/main/java/com/ptah/auth/AuthenticationService.java
package com.ptah.auth;

import com.ptah.entity.userprofiling.User;
import com.ptah.repository.userprofiling.UserRepository;
import com.ptah.dto.userprofiling.LoginResponse;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.stereotype.Service;


import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
public class AuthenticationService {
    private final UserRepository userRepository;
    private final JwtService jwtService;
    private final AuthenticationManager authenticationManager;

    public LoginResponse login(String login, String password) {
        authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(login, password));
        User user = userRepository.findByLogin(login).orElseThrow(() -> new IllegalArgumentException("Invalid email or password."));
        return LoginResponse.builder().accessToken(jwtService.generateToken(user)).build();
    }

}

File: backend/src/main/kotlin/com/github/jarryzhou/ptah/auth/AuthenticationService.kt
package com.github.jarryzhou.ptah.auth

import com.ptah.dto.userprofiling.LoginResponse
import com.ptah.repository.userprofiling.UserRepository
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.stereotype.Service

@Service
class AuthenticationService(
    var userRepository: UserRepository,
    var jwtService: JwtService,
    var authenticationManager: AuthenticationManager
) {
    fun login(login: String?, password: String?): LoginResponse {
        authenticationManager.authenticate(UsernamePasswordAuthenticationToken(login, password))
        val user = userRepository.findByLogin(login) ?: throw IllegalArgumentException("Invalid email or password.")
        return LoginResponse(accessToken = jwtService.generateToken(user))
    }
}
